data-products/src/main/scala/org/sunbird/analytics/sourcing/SourcingSummaryReport.scala

```diff
@@ -0,0 +1,129 @@
+import org.sunbird.cloud.storage.conf.AppConf
+
+case class PostgresData(programId: String, programName: String)
+case class SourcingContents(primaryCategory: String, createdBy: String, count: Int)
+
+object SourcingSummaryReport extends optional.Application with IJob with BaseReportsJob {
+
+  implicit val className = "org.sunbird.analytics.sourcing.SourcingSummaryReport"
+  val jobName: String = "SourcingSummaryReport"
+  val db = AppConf.getConfig("postgres.db")
+  val url = AppConf.getConfig("postgres.url") + s"$db"
+  val connProperties = CommonUtil.getPostgresConnectionProps
+  val tables = Map("programTable" -> "program", "nominationTable" -> "nomination","vUser" -> "\"V_User\"", "vUserOrg" -> "\"V_User_Org\"")
+
+  // $COVERAGE-OFF$ Disabling scoverage for main method
+  def main(config: String)(implicit sc: Option[SparkContext], fc: Option[FrameworkContext]): Unit = {
+    JobLogger.log(s"Started execution - $jobName",None, Level.INFO)
+    implicit val jobConfig = JSONUtils.deserialize[JobConfig](config)
+    implicit val frameworkContext: FrameworkContext = getReportingFrameworkContext()
+    implicit val spark = openSparkSession(jobConfig)
+
+    try {
+      val res = CommonUtil.time(execute())
+      JobLogger.end(s"$jobName completed execution", "SUCCESS", Option(Map("timeTaken" -> res._1)))
+    } finally {
+      frameworkContext.closeContext()
+      spark.close()
+    }
+  }
+
+  // $COVERAGE-ON$ Enabling scoverage for all other functions
+  def execute()(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig) = {
+    val programData = spark.read.jdbc(url, tables("programTable"), connProperties).withColumnRenamed("status","programStatus")
+    val nominationData = spark.read.jdbc(url, tables("nominationTable"), connProperties)
+    val projectDf = programData.join(nominationData, Seq("program_id"),"outer")
+      .select("program_id","status","rootorg_id","user_id")
+    JobLogger.log(s"Project count - ${projectDf.count()}",None, Level.INFO)
+
+    process(projectDf)
+  }
+
+  def process(projectDf: DataFrame)(implicit spark: SparkSession, fc: FrameworkContext, config: JobConfig) = {
+    val userDf = getUserDetails()
+    JobLogger.log(s"User data count - ${userDf.count()}",None, Level.INFO)
+    val resultDf = userDf.join(projectDf, userDf.col("userId") === projectDf.col("user_id"), "outer")
+      .withColumnRenamed("user_id","contributorId")
+      .withColumn("reportName", lit("SourcingReport"))
+      .withColumn("timestamp", lit(System.currentTimeMillis()))
+    JobLogger.log(s"resultDf count - ${resultDf.count()}",None, Level.INFO)
+
+    val modelParams = config.modelParams.get
+    val storageConfig = StorageConfig(AppConf.getConfig("cloud_storage_type"), "reports", "", Option(modelParams.getOrElse("storageKey","druid_storage_account_key").toString), Option(modelParams.getOrElse("storageSecret","druid_storage_account_secret").toString))
+    resultDf.saveToBlobStore(storageConfig, "json", "sourcing",
```

Comment:
you are mentioning file name as `sourcing` it's more generic. Can we rename this file name in the context of the job?

Reply:
Also, in the ingestion spec, the JSON file path is different, but it's uploading to different locations in the job, Please validate the path once.

Reply:
sourcing is the folder name where all the other reports for vdn are being stored. Have tested the changes, they are working as expected